lib/WUI/wui.cpp

```diff
@@ -0,0 +1,620 @@
+/*
+ * wui.h
+ * \brief main interface functions for Web User Interface (WUI) thread
+ *
+ *  Created on: Dec 12, 2019
+ *      Author: joshy
+  *  Modify on 09/17/2021
+ *      Author: Marek Mosna <marek.mosna[at]prusa3d.cz>
+*/
+
+#include "wui.h"
+#include "netif_settings.h"
+
+#include "marlin_client.h"
+#include "wui_api.h"
+#include "ethernetif.h"
+#include "espif.h"
+#include "stm32f4xx_hal.h"
+
+#include "sntp_client.h"
+#include "log.h"
+
+#include <atomic>
+#include <array>
+#include <cstring>
+#include <cassert>
+#include <lwip/ip.h>
+#include <lwip/dns.h>
+#include <lwip/tcp.h>
+#include <lwip/altcp_tcp.h>
+#include <lwip/netifapi.h>
+#include <lwip/netif.h>
+#include <lwip/tcpip.h>
+#include "http_lifetime.h"
+#include "main.h"
+
+#include "netdev.h"
+
+LOG_COMPONENT_DEF(WUI, LOG_SEVERITY_DEBUG);
+LOG_COMPONENT_DEF(Network, LOG_SEVERITY_INFO);
+
+// FIXME: " " vs <>
+#include "eeprom.h"
+#include "variant8.h"
+#include "otp.h"
+
+#define LOOP_EVT_TIMEOUT 500UL
+
+osMessageQDef(networkMbox, 16, NULL);
+osMessageQId networkMbox_id;
+
+static variant8_t prusa_link_api_key;
+
+const char *wui_generate_api_key(char *api_key, uint32_t length) {
+    // Avoid confusing character pairs ‒ 1/l/I, 0/O.
+    static char charset[] = "abcdefghijkmnopqrstuvwxyzABCDEFGHJKLMNPQRSTUVWXYZ23456789";
```

Comment:
This is not a new code (I don't know if github is capable of tracking renames in some config, in command line tools you might need to pass some kind of `--track-renames` ‒ at least, I've had to use some version of that for rebases to work well and had to google the exact invocation :-| ).

As for the suggestion to remove these too… I'm not sure. On one hand, they are slightly similar to each other, but not _that_ much. On the other hand, each removal makes the generated key weaker and we could continue with „V a Y look a bit similar“ and „M and N are almost the same` and `b/p/q/d` are very similar for some kind of people because they have the same shape, just mirrorred… and eventually end up with an empty list 😈.

Nevertheless, as it is pre-existing code, changing it is probably out of scope of this PR anyway. So if you believe some of these should be removed, would you take it up on some slack channel? We can always remove them after a decision is made.